module/apmgin/middleware.go

```diff
@@ -77,27 +73,17 @@ func (m *middleware) handle(c *gin.Context) {
 	}
 	m.setRouteMapOnce.Do(func() {
 		routes := m.engine.Routes()
-		rm := make(map[string]map[string]routeInfo)
+		m.routeMap = map[string]struct{}{}
 		for _, r := range routes {
-			mm := rm[r.Method]
-			if mm == nil {
-				mm = make(map[string]routeInfo)
-				rm[r.Method] = mm
-			}
-			mm[r.Handler] = routeInfo{
-				transactionName: r.Method + " " + r.Path,
-			}
+			m.routeMap[r.Method+" "+r.Path] = struct{}{}
 		}
-		m.routeMap = rm
 	})
 
-	var requestName string
-	handlerName := c.HandlerName()
-	if routeInfo, ok := m.routeMap[c.Request.Method][handlerName]; ok {
-		requestName = routeInfo.transactionName
-	} else {
+	requestName := c.Request.Method + " " + c.FullPath()
+	if _, ok := m.routeMap[requestName]; !ok {
```

Comment:
Building on top of the proposed changes, the solution can be further simplified by removing the `routeMap` completely and only check that the `fullPath` is not empty; the map does not provide any adavantage any more when not using the handler. 
Something similar to 

```
	var requestName string
	if fullPath := c.FullPath(); fullPath != "" {
		requestName = c.Request.Method + " " + c.FullPath()
	} else {
		requestName = apmhttp.UnknownRouteRequestName(c.Request)
	}
```